### PR TITLE
New: Throw error if fixture is not found

### PR DESCRIPTION
**Pull request by stramel**
- [x] CHANGELOG.md has been updated

Closes #338 since it had been sitting there for a while with just tests missing so I added some tests and fixed some code style issues with `function`

### CODE REVIEW

File: browser/mocha/fixture.js [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
     // test-fixture:
-    teardown(function () {
+    teardown(function() {
       document.getElementById(fixtureId).restore();
     });
 
     // Find the test-fixture with the provided ID and create it, returning
     // the results:
-    return document.getElementById(fixtureId).create(model);
+    var fixture = document.getElementById(fixtureId);
```
Comment (stramel) [inline browser/mocha/fixture.js]: Actual fix here

File: test/fixtures/integration/missing-fixture/golden.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (stramel) [inline test/fixtures/integration/missing-fixture/golden.json]: First of the new tests for changes.

**Comment by stramel** [comment]
Rebased onto master with green build, changelog, tests, and browser.js ref fix.

**Comment by usergenic** [comment]
Merging to a branch to try and fix CI/tests first.

**Pull request merged**